HLTrigger/HLTfilters/plugins/HLTDoubletDZ.cc

```diff
@@ -431,17 +430,15 @@ bool HLTDoubletDZ<reco::RecoChargedCandidate, reco::RecoChargedCandidate>::compu
 }
 
 template <>
-bool HLTDoubletDZ<l1t::TkMuon, l1t::TkMuon>::computeDZ(edm::Event& iEvent,
-                                                       l1t::TkMuonRef& r1,
-                                                       l1t::TkMuonRef& r2) const {
-  const l1t::TkMuon& candidate1(*r1);
-  const l1t::TkMuon& candidate2(*r2);
-  if (reco::deltaR(candidate1, candidate2) < minDR_)
+bool HLTDoubletDZ<l1t::TrackerMuon, l1t::TrackerMuon>::computeDZ(edm::Event& iEvent,
+                                                                 l1t::TrackerMuonRef& r1,
+                                                                 l1t::TrackerMuonRef& r2) const {
+  if (reco::deltaR2(r1->phEta(), r1->phPhi(), r2->phEta(), r2->phPhi()) < minDR_ * minDR_)
     return false;
 
-  // We don't care about minPixHitsForDZ_ with the L1TkMuons,
+  // We don't care about minPixHitsForDZ_ with the L1TTrackerMuons,
   // especially because the pixel does not participate in the L1T
-  if (std::abs(candidate1.trkzVtx() - candidate2.trkzVtx()) > maxDZ_)
+  if (std::abs(r1->phZ0() - r2->phZ0()) > maxDZ_)
```

Comment:
This computation is less expensive than the one above, it may be worth moving it before that